EpiAware/src/EpiLatentModels/odemodels/SEIRParams.jl

```diff
@@ -0,0 +1,175 @@
+    infectiousness::D
+    "Prior distribution for the incubation rate parameter."
+    incubation_rate::E
+    "Prior distribution for the recovery rate parameter."
+    recovery_rate::F
+    "Prior distribution for initial proportion of the population that is infected."
+    initial_prop_infected::G
+end
+
+function SEIRParams(;
+        tspan, infectiousness::Distribution, incubation_rate::Distribution,
+        recovery_rate::Distribution, initial_prop_infected::Distribution)
+    seir_prob = ODEProblem(_seir_function, [0.99, 0.05, 0.05, 0.0], tspan)
+    return SEIRParams{
+        typeof(seir_prob), typeof(infectiousness), typeof(incubation_rate),
+        typeof(recovery_rate), typeof(initial_prop_infected)}(
+        seir_prob, infectiousness, incubation_rate,
+        recovery_rate, initial_prop_infected)
+end
+
+@doc raw"""
+Generates the initial parameters and initial conditions for the basic SEIR model.
+
+## SEIR model
+
+```math
+\begin{aligned}
+\frac{dS}{dt} &= -\beta SI \\
+\frac{dE}{dt} &= \beta SI - \alpha E \\
+\frac{dI}{dt} &= \alpha E - \gamma I \\
+\frac{dR}{dt} &= \gamma I
+\end{aligned}
+```
+Where `S` is the proportion of the population that is susceptible, `E` is the proportion of the
+population that is exposed, `I` is the proportion of the population that is infected and `R` is
+the proportion of the population that is recovered. The parameters are the infectiousness `β`,
+the incubation rate `α` and the recovery rate `γ`.
+
+## Initial conditions
+
+For this version of the SEIR model we sample the initial proportion of the population that is
+_infected_ (exposed or infectious). The proportion of the infected group that is exposed is
+`α / (α + γ)` and the proportion of the infected group that is infectious is `γ / (α + γ)`. The
+reason for this is that these are the equilibrium proportions in a constant incidence environment.
+
+# Example
+
+```julia
+using EpiAware, OrdinaryDiffEq, Distributions
+
+# Create an instance of SIRParams
+seirparams = SEIRParams(
+    tspan = (0.0, 30.0),
+    infectiousness = LogNormal(log(0.3), 0.05),
+    incubation_rate = LogNormal(log(0.1), 0.05),
+    recovery_rate = LogNormal(log(0.1), 0.05),
+    initial_prop_infected = Beta(1, 99)
+)
+
+seirparam_mdl = generate_latent(seirparams, nothing)
+
+# Sample the parameters of SEIR model
+sampled_params = rand(seirparam_mdl)
+```
+
+# Returns
+  A tuple `(u0, p)` where:
+  - `u0`: A vector representing the initial state of the system `[S₀, E₀, I₀, R₀]` where `S₀`
+  is the initial proportion of susceptible individuals, `E₀` is the initial proportion of exposed
+  individuals,`I₀` is the initial proportion of infected individuals, and `R₀` is the initial
+  proportion of recovered individuals.
+  - `p`: A vector containing the parameters `[β, α, γ]` where `β` is the infectiousness rate,
+  `α` is the incubation rate, and `γ` is the recovery rate.
+"""
+@model function EpiAwareBase.generate_latent(params::SEIRParams, Z_t)
```

Comment:
The implicit promise of this being a latent model is that it will work with the library of latent model functionality. Does this implementation do so? If no do we need to rethink the idea of making this a latent model

Reply:
I'm not sure off the top of my head... I think this an example of something we should think about generally #525